src/components/motion/MotionController.cpp

```diff
@@ -14,69 +14,87 @@ void MotionController::Update(int16_t x, int16_t y, int16_t z, uint32_t nbSteps)
   this->x = x;
   this->y = y;
   this->z = z;
-  int32_t deltaSteps = nbSteps - this->nbSteps;
+  uint32_t deltaSteps = nbSteps - this->nbSteps;
```

Comment:
This shouldn't be unsigned, otherwise the check below will always return true.

Reply:
I was attempting best practices since I was given a warning from clang-tidy, but I will revert it for now until I get a better understanding of what's going on.